src/Avalonia.X11/X11Window.cs

```diff
@@ -135,7 +135,7 @@ public X11Window(AvaloniaX11Platform platform, IWindowImpl popupParent)
             else
                 _renderHandle = _handle;
                 
-            Handle = new PlatformHandle(_handle, "XID");
+            Handle = new SurfacePlatformHandle(_handle, "XID", this);
```

Comment:
You need to use `_renderHandle`. In some cases we set up a child window to render into.

Reply:
Addressed